CMakeLists.txt

```diff
@@ -1,23 +1,35 @@
 CMAKE_MINIMUM_REQUIRED(VERSION 2.6 FATAL_ERROR)
 CMAKE_POLICY(VERSION 2.6)
+ENABLE_TESTING()
 
 # This CMake file is used _only_ really to install torchcraft in lua
 IF(LUAROCKS_PREFIX)
     MESSAGE(STATUS "Installing TorchCraft through Luarocks")
     STRING(REGEX REPLACE "(.*)lib/luarocks/rocks.*" "\\1" CMAKE_INSTALL_PREFIX  "${LUAROCKS_PREFIX}")
     MESSAGE(STATUS "Prefix inferred from Luarocks: ${CMAKE_INSTALL_PREFIX}")
 ENDIF()
+IF(NOT MSVC)
+    ADD_SUBDIRECTORY(test)
+ENDIF(NOT MSVC)
 FIND_PACKAGE(Torch REQUIRED)
 FIND_PACKAGE(PkgConfig REQUIRED)
 PKG_CHECK_MODULES(ZSTD libzstd>=1.0)
 
-SET(CMAKE_CXX_FLAGS "-std=c++11 -Wall -fPIC -O2")
+SET(CMAKE_CXX_STANDARD 14)
+SET(CMAKE_CXX_STANDARD_REQUIRED ON)
+SET(CMAKE_CXX_FLAGS_DEBUG "-O0 -g -ggdb")
```

Comment:
Remember to remove this! Also I think you can just do `-DCMAKE_BUILD_TYPE=Debug` to cmake.

Reply:
I've removed the temporary instruction to always build as debug; this instruction will cause cmake to use the flags only when someone opts to build the debug version.

Reply:
It would be good practice to only extend the debug flags with things you think are required. However, since `-g` is already defined by default for the debug flags and I'm not sure why you need `-ggdb`, maybe just do this: `SET(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -O0")`